gramps_webapi/api/resources/delete.py

```diff
@@ -0,0 +1,414 @@
+            db_handle.commit_event(event, trans)
+
+        for _handle in place_list:
+            place = db_handle.get_place_from_handle(_handle)
+            place.remove_citation_references(ctn_handle_list)
+            db_handle.commit_place(place, trans)
+
+        for _handle in source_list:
+            source = db_handle.get_source_from_handle(_handle)
+            source.remove_citation_references(ctn_handle_list)
+            db_handle.commit_source(source, trans)
+
+        for _handle in media_list:
+            media = db_handle.get_media_from_handle(_handle)
+            media.remove_citation_references(ctn_handle_list)
+            db_handle.commit_media(media, trans)
+
+        for _handle in repo_list:
+            repo = db_handle.get_repository_from_handle(_handle)
+            repo.remove_citation_references(ctn_handle_list)
+            db_handle.commit_repository(repo, trans)
+
+    for citation_handle in citation_list:
+        db_handle.remove_citation(citation_handle, trans)
+
+    db_handle.remove_source(handle, trans)
+
+
+def delete_tag(db_handle: DbWriteBase, handle: str, trans: DbTxn) -> None:
+    """Delete a tag."""
+    fnc = {
+        "Person": (db_handle.get_person_from_handle, db_handle.commit_person),
+        "Family": (db_handle.get_family_from_handle, db_handle.commit_family),
+        "Event": (db_handle.get_event_from_handle, db_handle.commit_event),
+        "Place": (db_handle.get_place_from_handle, db_handle.commit_place),
+        "Source": (db_handle.get_source_from_handle, db_handle.commit_source),
+        "Citation": (db_handle.get_citation_from_handle, db_handle.commit_citation),
+        "Repository": (
+            db_handle.get_repository_from_handle,
+            db_handle.commit_repository,
+        ),
+        "Media": (db_handle.get_media_from_handle, db_handle.commit_media),
+        "Note": (db_handle.get_note_from_handle, db_handle.commit_note),
+    }
+    links = db_handle.find_backlink_handles(handle)
+
+    for classname, _handle in links:
+        obj = fnc[classname][0](_handle)  # get from handle
+        obj.remove_tag(handle)
+        fnc[classname][1](obj, trans)  # commit
+    db_handle.remove_tag(handle, trans)
+
+
+delete_methods = {
+    "person": delete_person,
+    "family": delete_family,
+    "event": delete_event,
+    "place": delete_place,
+    "media": delete_media,
+    "note": delete_note,
+    "repository": delete_repository,
+    "source": delete_source,
+    "citation": delete_citation,
+    "tag": delete_tag,
+}
+
+
+def delete_object(
+    db_handle: DbWriteBase, handle: str, gramps_class_name: str
+) -> List[Dict[str, Any]]:
+    """Delete the object and its references."""
+    key = gramps_class_name.lower()
+    try:
+        method = delete_methods[key]
+    except KeyError:
+        raise NotImplementedError(gramps_class_name)
+    with DbTxn(f"Delete {gramps_class_name} {handle}", db_handle) as trans:
```

Comment:
This message won't be exposed to the API user at any time, so if at all, it might appear in a debug log; in that case, the handle might even be better as it's guaranteed to be unique.

Reply:
Okay